src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java

```diff
@@ -45,8 +45,8 @@ public FirebaseDataStorage() {
     // Create a query to find a macro named macroName belonging to userEmail
     Query query =
         db.collection(COLLECTION_NAME)
-            .whereEqualTo(USER_IDENTIFIER, userEmail)
-            .whereEqualTo(MACRO_IDENTIFIER, macroName);
+            .whereEqualTo(MACRO_IDENTIFIER, macroName)
+            .whereArrayContains(SCOPE_IDENTIFIER, userEmail);
```

Comment:
do we want to allow the case in which the user leaves scope empty and then we allow all users? Or in this case only the user should be able to use the macro? Maybe we can use some kind of keywords for it like "ALL" or "PERSONAL"